### DISCUSSION TITLE
raspberrypi Script list for dropping and jitter. And how to use it.

### DISCUSSION THREAD

**Discussion by codekistone** [changjurhee/AONE#24] [General]
**[AddDrop.sh]**
sudo iptables -A FORWARD  -m statistic --mode random --probability 0.10  -j DROP

**[AddDrop_DemoDay.sh]**
sudo iptables -A FORWARD -m statistic --mode random --probability 0.05 -j DROP
sudo iptables -A OUTPUT -m statistic --mode random --probability 0.05 -j DROP

**[RemoveDrop.sh]**
sudo iptables -F
sudo iptables -X

---

**[AddJitter.sh]**
sudo tc qdisc add dev wlan0  root netem delay 300ms 50ms
sudo tc qdisc add dev wlan1  root netem delay 300ms 50ms

**[AddJitter_DemoDay.sh]**
sudo tc qdisc add dev wlan0  root netem delay 10ms 20ms distribution normal rate 10Gbit
sudo tc qdisc add dev wlan1  root netem delay 10ms 20ms distribution normal rate 10Gbit

**[AddJitter_new.sh]**
sudo tc qdisc add dev wlan0  root netem delay 300ms 50ms distribution normal rate 10Gbit
sudo tc qdisc add dev wlan1  root netem delay 300ms 50ms distribution normal rate 10Gbit

**[RemoveJitter.sh]**
sudo tc qdisc del dev wlan0 root
sudo tc qdisc del dev wlan1 root

**Comment by junsoopark** [comment]
https://github.com/changjurhee/AONE/issues/33
> You connect one machine to network one, and you connect the second machine to network 2, and then. you know. your applications will communicate to each other. Okay, via these 2 networks. But just one application.

테스트하고자 하는 2개의 디바이스를 서로다른 AP에 연결하고 jitter, drop 설정 테스트.

**Comment by junsoopark** [comment]
https://github.com/changjurhee/AONE/issues/39
패킷 드랍을 위한 iptables 명령어 사용법.

https://linuxstory1.tistory.com/entry/iptables-%EA%B8%B0%EB%B3%B8-%EB%AA%85%EB%A0%B9%EC%96%B4-%EB%B0%8F-%EC%98%B5%EC%85%98-%EB%AA%85%EB%A0%B9%EC%96%B4

**Comment by junsoopark** [comment]
지터 발생을 위한 명령어 사용법.

https://man.archlinux.org/man/core/iproute2/tc-netem.8.en
https://www.cs.unm.edu/~crandall/netsfall13/TCtutorial.pdf
[NetEM을 이용한 Traffic Shaping (패킷 지연,유실,대역폭 제한등)](http://www.packetinside.com/2013/01/netem-traffic-shaping.html?m=1)
